packages/formatter-json/src/formatter.ts

```diff
@@ -34,17 +35,30 @@ const debug = d(__filename);
 
 export default class JSONFormatter implements IFormatter {
     /** Format the problems grouped by `resource` name and sorted by line and column number */
-    public format(messages: Problem[]) {
+    public async format(messages: Problem[], target: string | undefined, options: FormatterOptions = {}) {
```

Comment:
@antross @sarvaje what do you think about combining `target` and `options`? `target` is not used by all the `formatters` but `options` yes for the filename output so right now is a bit weird

Reply:
I like the idea

Reply:
@molant done.